students/daniel_carrasco/session09/donor_database.py

```diff
@@ -0,0 +1,47 @@
+class Donor_Database:
+    data = {}
+
+    def __init__(self, name=""):
+        self.name = name
+        if name != "":
+            try:
+                self.data[name]
+            except KeyError:
+                self.data.update({name: [0, 0]})
+
+    def donations(self, amount=0):
+        current_donation = self.data[self.name][0]
+        current_count = self.data[self.name][1]
+        if amount == 0:
+            self.data.update(
+                {self.name: [amount + current_donation, 1]})
```

Comment:
`list` objects that you have here have a length you can derive by calling `len()` on them, so you don't need to keep a count of donations as part of your data structure.

Reply:
example:
```
>>> donations = {"john": [123.00, 3123.02, 112.00]}
>>> len(donations["john"])
3
```